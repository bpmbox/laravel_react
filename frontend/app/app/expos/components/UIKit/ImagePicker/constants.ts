/**
 * Shared constants used by the image picker on all platforms.
 */

export const RESIZE_MAX_SIZE = 1024;
