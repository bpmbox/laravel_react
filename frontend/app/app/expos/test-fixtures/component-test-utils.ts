/* istanbul ignore file */
import { SinonFakeTimers, SinonSandbox } from 'sinon';
import { act } from 'react-dom/test-utils';
import alertService from '../services/alert';

export async function awaitPromises(clock: SinonFakeTimers | null = null, waitTime: number = 50) {
    // Run an act cycle to prevent logging warnings.
    act(() => {});

    await new Promise(resolve => {
        // Note: this shows as a squiggly warning in VSCode, however passes.  'setImmediate' is a global provided by NodeJS
        setImmediate(() => {
            resolve();
        });

        // If we are using a fake timer, we must advance the clock for promises to resolve.
        // Note: this is done after the setImmediate so the act of doing this will trigger the above promise.
        if (clock) {
            clock.tick(waitTime);
        }
    });
}

export function mockAlert(sandbox: SinonSandbox) {
    const alertStub = sandbox.stub(alertService, 'alert');

    const confirmAlert = () => {
        const handlerFunc = alertStub.lastCall.args[2][1].onPress as CallableFunction;
        handlerFunc();
    };

    return {
        alertStub,
        confirmAlert,
    };
}
