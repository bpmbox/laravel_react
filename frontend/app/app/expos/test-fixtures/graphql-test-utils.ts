/* istanbul ignore file */
import { SinonStub } from 'sinon';
import { ApolloClient } from 'apollo-client';
import { InMemoryCache, NormalizedCacheObject } from 'apollo-cache-inmemory';
import { MockLink } from '@apollo/react-testing';

/**
 * Gets the name of the first mutation called on a sinon stub of graphqlService.mutate.
 * @param graphqlStub
 * @param call Call index of mocked function.
 */
export function getFirstMutation(mutateStub: SinonStub, call = 0) {
    return mutateStub.getCall(call).args[0].mutation.definitions[0].selectionSet.selections[0].name.value;
}

/**
 * Tests if a mutation contains a String or regex.
 * @param mutateStub
 * @param matcher
 * @param call
 */
export function mutationContains(mutateStub: SinonStub, matcher: RegExp | string, call: number = 0) {
    const mutationAsString = JSON.stringify(mutateStub.getCall(call).args[0]);

    return mutationAsString.search(matcher) >= 0;
}

/**
 * Creates a test Apollo client with support for mock responses
 * @param mocks mock requests and responses
 * @param addTypename add `__typename` field for queries (default: true)
 * @returns ApolloClient
 */
export function createTestClient(mocks: any[], addTypename = true): ApolloClient<NormalizedCacheObject> {
    return new ApolloClient({
        link: new MockLink(mocks || [], addTypename),
        cache: new InMemoryCache({ addTypename }),
    });
}
