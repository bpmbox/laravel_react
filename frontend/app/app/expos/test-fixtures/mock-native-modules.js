/**
 * Mock Native moodules, so they can run in NodeJS env for unit tests.
 */
import { NativeModules } from 'react-native';

jest.mock('react-native-fs', () => {
    return {
        DocumentDirectoryPath: '/test',
    };
});

jest.mock('react-native-device-info', () => {
    return {
        isEmulatorSync: () => false,
    };
});

jest.mock('react-native-file-viewer', () => {
    return {
        open: jest.fn().mockResolvedValue(null),
    };
});

jest.mock('@react-native-community/async-storage', () => {
    let cache = {};
    return {
        setItem: (key, value) => {
            return new Promise((resolve, reject) => {
                return typeof key !== 'string' || typeof value !== 'string'
                    ? reject(new Error('key and value must be string'))
                    : resolve((cache[key] = value));
            });
        },
        getItem: (key, value) => {
            return new Promise(resolve => {
                return cache.hasOwnProperty(key) ? resolve(cache[key]) : resolve(null);
            });
        },
        removeItem: key => {
            return new Promise((resolve, reject) => {
                return cache.hasOwnProperty(key) ? resolve(delete cache[key]) : reject('No such key!');
            });
        },
        clear: key => {
            return new Promise((resolve, reject) => resolve((cache = {})));
        },
        getAllKeys: key => {
            return new Promise((resolve, reject) => resolve(Object.keys(cache)));
        },
    };
});

jest.mock('@invertase/react-native-apple-authentication', () => {
    return {
        AppleAuth: {
            getCredentialStateForUser: _args => {
                return Promise.resolve({});
            },
            performRequest: _args => {
                return Promise.resolve({});
            },
        },

        AppleAuthCredentialState: {
            APPROVED: '1',
            REVOKED: '2',
        },

        AppleAuthError: {
            CANCELED: '1001',
            INVALID_RESPONSE: '1002',
        },

        AppleAuthRequestScope: {
            EMAIL: '3',
            FULL_NAME: '4',
        },
    };
});

jest.mock('@react-native-community/google-signin', () => {
    return {
        GoogleSignin: {
            configure: () => Promise.resolve(),

            hasPlayServices: () => Promise.resolve(true),

            currentUserAsync: () => {
                return Promise.resolve({
                    name: 'name',
                    email: 'test@example.com',
                    // .... other user data
                });
            },
        },
    };
});

NativeModules.RNGoogleSignin = {
    BUTTON_SIZE_ICON: 0,
    BUTTON_SIZE_STANDARD: 0,
    BUTTON_SIZE_WIDE: 0,
    BUTTON_COLOR_AUTO: 0,
    BUTTON_COLOR_LIGHT: 0,
    BUTTON_COLOR_DARK: 0,
    SIGN_IN_CANCELLED: '0',
    IN_PROGRESS: '1',
    PLAY_SERVICES_NOT_AVAILABLE: '2',
    SIGN_IN_REQUIRED: '3',
    configure: jest.fn(),
    currentUserAsync: jest.fn(),
};

// Pulled from https://github.com/react-native-community/react-native-image-picker/blob/master/jest.setup.js
NativeModules.ImagePickerManager = {
    showImagePicker: jest.fn(),
    launchCamera: jest.fn(),
    launchImageLibrary: jest.fn(),
};

// Pulled from https://github.com/invertase/react-native-firebase/issues/1902#issuecomment-480886526
jest.mock('react-native-firebase', () => ({
    messaging: jest.fn(() => ({
        hasPermission: jest.fn(() => Promise.resolve(true)),
        subscribeToTopic: jest.fn(),
        unsubscribeFromTopic: jest.fn(),
        requestPermission: jest.fn(() => Promise.resolve(true)),
        getToken: jest.fn(() => Promise.resolve('myMockToken')),
    })),
    notifications: jest.fn(() => ({
        onNotification: jest.fn(),
        onNotificationDisplayed: jest.fn(),
    })),
    analytics: jest.fn(() => ({
        logEvent: jest.fn(),
    })),
}));

jest.mock('react-native-reanimated', () => require('react-native-reanimated/mock'));

/**
 * Poly fill fetch() for unit tests.  Because react native does not have global fetch
 * when run in NodeJS.
 */
require('jest-fetch-mock').enableMocks();
