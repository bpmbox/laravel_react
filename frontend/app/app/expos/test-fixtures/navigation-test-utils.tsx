/* istanbul ignore file */
import { render } from '@testing-library/react-native';
import React from 'react';
import { createAppContainer } from 'react-navigation';
import { HistoryService } from '../services/history';
import AuthStore from '../store/auth';

/**
 * Test utility for rendering navigation stack for react native.
 * @param historyService
 * @param navigatorInstance
 * @param initialAuthState
 */
export function renderNavigation(historyService: HistoryService, navigatorInstance: any, initialAuthState?: any) {
    const App = createAppContainer(navigatorInstance);

    const authState = Object.assign({}, initialAuthState, {
        isAuthenticated: false,
        currentUser: null,
        isLoading: false,
    });

    const tree = render(
        <AuthStore.Provider initialState={authState}>
            <App ref={navigatorRef => historyService.setNavigation(navigatorRef)} />
        </AuthStore.Provider>
    );

    return tree;
}
