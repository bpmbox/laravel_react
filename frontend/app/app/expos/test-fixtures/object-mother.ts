/* istanbul ignore file */
import { ApolloError } from 'apollo-client';
import { ASTNode } from 'graphql';
import { DateTime } from 'luxon';
import { sign } from 'jsonwebtoken';
import { Role } from '../types/enums';

/**
 * Object Mother for frequently reused test data.
 */

export const simpleApolloError = new ApolloError({
    graphQLErrors: [
        {
            message: 'test error',
            locations: [{ line: 1, column: 10 }],
            path: ['Space', 'id'],
            nodes: [{} as ASTNode],
            source: undefined,
            positions: [5],
            originalError: null,
            name: 'SomethingWrongError',
            extensions: {
                code: '001-500-0011',
                time_thrown: DateTime.local().toISO(),
            },
        },
    ],
    networkError: null,
});

export const sampleUser: User = {
    id: 'abc123',
    givenName: 'Jon',
    familyName: 'Snow',
    email: 'j.snow@winterfell.com',
    avatarUrl:
        'https://vignette.wikia.nocookie.net/gameofthrones/images/d/d0/JonSnow8x06.PNG/revision/latest?cb=20190714094440',
};

export const sampleGroup: Group = {
    id: '1111111-1111-1111-111111111111',
    name: 'Sample group',
};

export const sampleSpace: Space = {
    id: '112233',
    name: 'Space One',
    slug: 'space-one',
    iconUrl: 'http://icons.iconarchive.com/icons/limav/game-of-thrones/512/Baratheon-icon.png',
};

export const sampleSpaceInfo: SpaceServiceTypes.SpaceInfo = {
    space: sampleSpace,
    role: Role.ADMIN,
    chat: {
        userId: '123',
        accessToken: 'abc',
    },
    invited: false,
};

export const sampleIntegration: NSIntegration.Integration = {
    id: '1111111-1111-1111-111111111111',
    name: 'Event',
    logo: 'https://s3.amazonaws.com/dev-uploads.withtree.com/tmp/test.jpg',
    shortDesc: 'This is event',
    fullDesc: 'This is full desc of event',
    category: 'fun',
    type: 'native',
    url: 'http://test.com',
    access: 'published',
    permissions: ['can_view', 'full_access'],
    restrictedSpaces: [],
};

export const sampleNewIntegration: NSIntegration.NewIntegrationParams = {
    name: 'Event',
    logo: 'https://s3.amazonaws.com/dev-uploads.withtree.com/tmp/test.jpg',
    shortDesc: 'This is event',
    fullDesc: 'This is full desc of event',
    category: 'fun',
    type: 'native',
    url: 'http://test.com',
    access: 'published',
    permissions: ['can_view', 'full_access'],
    restrictedSpaceSlugs: [],
};

export const sampleIntegrationInfo: NSIntegration.IntegrationInfo = {
    integration: sampleIntegration,
    isInstalled: true,
    isActive: true,
    installationId: '1111111-1111-1111-111111111111',
    isHomepage: false,
};

export const sampleAvailableIntegrationInfo: NSIntegration.IntegrationInfo = {
    integration: sampleIntegration,
    isInstalled: false,
    isActive: false,
    installationId: '',
    isHomepage: false,
};

export const sampleInstallationInfo: NSIntegration.InstallationInfo = {
    usersPermissions: [{ user: sampleUser, permissions: ['can_view'] }],
    groupsPermissions: [{ group: sampleGroup, permissions: ['can_view'] }],
};

export const sampleSpaces: Space[] = [
    {
        id: '112233',
        name: 'Space One',
        slug: 'space-one',
        iconUrl:
            'https://static.tumblr.com/8f24acdc4b08ac20a2de3b8378b28934/eomrklq/JFDncmaga/tumblr_static_3wev7wkhxa0440s4g04wcc840.png',
    },
    {
        id: '223344',
        name: 'Space Two',
        slug: 'space-two',
        iconUrl: 'http://icons.iconarchive.com/icons/limav/game-of-thrones/512/Baratheon-icon.png',
    },
];

export const sampleMembersWithChat: SpaceServiceTypes.MembershipWithChat[] = [
    {
        member: {
            id: 'abc123',
            givenName: 'Jon',
            familyName: 'Snow',
            email: 'j.snow@winterfell.com',
            avatarUrl:
                'https://vignette.wikia.nocookie.net/gameofthrones/images/d/d0/JonSnow8x06.PNG/revision/latest?cb=20190714094440',
        },
        role: Role.OWNER,
        chat: {
            userId: 'abc123',
            accessToken: 'SECRET',
        },
    },
    {
        member: {
            id: 'def456',
            givenName: 'Daenerys',
            familyName: 'Targarian',
            email: 'd.targarian@dragonriders.com',
            avatarUrl:
                'https://www.telegraph.co.uk/content/dam/fashion/2017/06/19/TELEMMGLPICT000131421802_trans_NvBQzQNjv4BqkUE_BTgBOQu3VWKvpDGX9fr7sARQy7EgBjwPUzvqL_M.jpeg?imwidth=450',
        },
        role: Role.ADMIN,
        chat: {
            userId: 'def456',
            accessToken: 'SECRET',
        },
    },
    {
        member: {
            id: 'ghi789',
            givenName: 'Tyrion',
            familyName: 'Lannister',
            email: 't.lannister@kingslanding.com',
            avatarUrl: 'https://static.toiimg.com/photo/62981209/.jpg',
        },
        role: Role.MEMBER,
        chat: {
            userId: 'ghi789',
            accessToken: 'SECRET',
        },
    },
    {
        member: {
            id: 'jkl101112',
            givenName: 'Grey',
            familyName: 'Worm',
            email: 'gworm@unsullied.com',
            avatarUrl: 'https://www.stickpng.com/assets/images/5cb83a4a32c4f52f2e5ea223.png',
        },
        role: Role.MEMBER,
        chat: {
            userId: 'jkl101112',
            accessToken: 'SECRET',
        },
    },
    {
        member: {
            id: 'mno131415',
            givenName: 'Arya',
            familyName: 'Stark',
            email: 'as@faceless.com',
            avatarUrl: 'https://pbs.twimg.com/profile_images/1117701700360581126/4p020QBx_400x400.jpg',
        },
        role: Role.MEMBER,
        chat: {
            userId: 'mno131415',
            accessToken: 'SECRET',
        },
    },
];

export const sampleGroups: Group[] = [
    { id: '1', name: 'Los Angeles' },
    { id: '2', name: 'Hanoi' },
    { id: '5', name: 'Paris' },
    { id: '6', name: 'Tokyo' },
    { id: '7', name: 'London' },
    { id: '8', name: 'New York' },
    { id: '9', name: 'Cairo' },
];

export const sampleAccessToken: string = sign(sampleUser, 'secret', {
    expiresIn: '1h',
});

export const sampleRefreshToken: string = sign(sampleUser, 'secret', {
    expiresIn: '4320h', // 6 months
});
