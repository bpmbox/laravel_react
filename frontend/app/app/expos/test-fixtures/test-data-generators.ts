/* istanbul ignore file */
import faker from 'faker';

/**
 * Creates an iterator of that returns fake Users.
 *
 * Usage:
 *      const fakeUsers = Array.from(generateUser(5));
 * @param qty number of items to generate
 */
export function* generateUser(qty: number) {
    for (let i = 0; i < qty; i++) {
        yield {
            id: faker.random.uuid(),
            givenName: faker.name.firstName(),
            familyName: faker.name.lastName(),
            email: faker.internet.email(),
            avatarUrl: faker.image.avatar(),
        };
    }
}

/**
 * Creates an iterator that returns fake Groups
 *
 * Usage:
 *      const groups = Array.from(generateGroup(5));
 * @param qty number of items to generate
 */
export function* generateGroup(qty: number) {
    for (let i = 0; i < qty; i++) {
        yield {
            id: faker.random.uuid(),
            name: `${faker.company.bsNoun()} ${faker.commerce.department}`,
        };
    }
}
