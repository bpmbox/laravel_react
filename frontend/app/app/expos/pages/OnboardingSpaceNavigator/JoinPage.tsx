import { withNavigation } from '@react-navigation/core';
import React, { useEffect, useRef, useState } from 'react';
import { useTranslation } from 'react-i18next';
import Button from '../../components/UIKit/items/Button';
import SectionHeading from '../../components/UIKit/items/SectionHeading';
import Spacer from '../../components/UIKit/items/Spacer';
import SpaceSelectorItem from '../../components/UIKit/items/SpaceSelectorItem';
import Text from '../../components/UIKit/items/Text';
import Page from '../../components/UIKit/Layout/Page';
import Spinner from '../../components/UIKit/Spinner';
import i18n from '../../i18n';
import { PARAM_SPACE } from '../../constants';
import routes from '../../routes';
import { ButtonType, ItemWidth } from '../../theme.style';
import authService from '../../services/auth';
import historyService from '../../services/history';
import messageService from '../../services/message';
import spaceService from '../../services/space';
import AuthStore from '../../store/auth';
import { isMobilePlatform } from '../../libs/platform';

const JoinPage = (props: { navigation: any }) => {
    const { navigation } = props;
    const { t } = useTranslation('OnboardingJoinSpacePage');
    const { currentUser } = AuthStore.useContainer();
    const [invitedSpaces, setInvitedSpaces] = useState<Space[]>([]);
    const [isJoining, setIsJoining] = useState<boolean>(false);
    const joiningSpaceRef = useRef<Space>(); // use to track last selected space to join.

    const count = (invitedSpaces && invitedSpaces.length) || 0;

    useEffect((): void => {
        (async function fetchSpaces(): Promise<void> {
            try {
                const spaces = await spaceService.getInvitedSpaces();
                setInvitedSpaces(spaces);
            } catch (e) {
                messageService.sendError(e.message, t`Error`);
            }
        })();
        // eslint-disable-next-line react-hooks/exhaustive-deps
    }, []);


    const getDataandJoin = async (space: Space) => {
        if (isMobilePlatform){
            //モバイルアプリの場合はチュートリアル画面へ遷移
            navigation.navigate(routes.ONBOARDING_SPACE_INTRO, {[PARAM_SPACE]: space});
        }else{
            joinSpace(space);
        }
    }

    const joinSpace = async (space: Space) => {    
        joiningSpaceRef.current = space;
        setIsJoining(true);
        try {
            await spaceService.joinSpace(space);
            setIsJoining(false);
            navigation.navigate(routes.MAIN_SPACE_REDIRECT, {
                [PARAM_SPACE]: space,
            });
        } catch (e) {
            console.error('Error occurred while trying to join space', e);
            messageService.sendError(
                t('Error occurred while trying to join {{spaceName}}.', {
                    spaceName: space.name,
                })
            );
            setIsJoining(false);
        }
    };

    const handleCreateSpace = (): void => {
        historyService.push(routes.ONBOARDING_SPACE_CREATE);
    };

    const handleLogOut = async () => {
        try {
            await authService.logout();
            historyService.navigateAsRoot(routes.LOGIN_HOME);
        } catch (err) {
            messageService.sendError(t`Error logging out.`);
        }
    };

    return <Page scrollable>
        {count === 0 &&
            <>
                <Text
                    text={t`A space is where your community comes to life. You can create or join other spaces later.`}
                    light
                    small
                    desktopWidth={ItemWidth.narrow}
                    desktopCenterItem={true}
                    />
                <Spacer />
                <Button
                    onPress={handleCreateSpace}
                    text={t`Create a Space`}
                    desktopWidth={ItemWidth.narrow}
                    desktopCenterItem={true}/>
                <Button
                    onPress={handleLogOut}
                    text={t('Switch Account')}
                    type={ButtonType.text}
                    desktopWidth={ItemWidth.narrow}
                    desktopCenterItem={true}/>
            </>
        }

        {count > 0 &&
            <>
                <Button
                    onPress={handleLogOut}
                    text={t('Switch Account')}
                    desktopWidth={ItemWidth.narrow}
                    desktopCenterItem={true}/>
                <SectionHeading
                    text={t`Spaces you've been invited to`}
                    desktopWidth={ItemWidth.narrow}
                    desktopCenterItem={true}
                />
                {isJoining
                    ? <Spinner />
                    : invitedSpaces.map((space: Space) => {
                        return <SpaceSelectorItem
                            testID={`SelectSpace_${space.id}`}
                            space={space}
                            key={space.slug}
                            onPress={() => { getDataandJoin(space) }}
                            desktopWidth={ItemWidth.narrow}
                            desktopCenterItem={true}
                        />
                    })
                }
            </>
        }

        <Spacer/>

        {currentUser && currentUser.email &&
            <Text
                text={t('Using account: {{email}}', {
                    email: currentUser.email,
                })}
                light
                small
                desktopWidth={ItemWidth.narrow}
                desktopCenterItem={true}
            />
        }
    </Page>;
};

JoinPage.navigationOptions = {
    title: i18n.t('JoinPage::Add a Space'),
};

// @ts-ignore
JoinPage.path = 'join'; //override path for better web URLs

export default withNavigation(JoinPage);
