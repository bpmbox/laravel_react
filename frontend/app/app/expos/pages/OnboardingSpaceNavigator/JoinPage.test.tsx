import { act } from '@testing-library/react-hooks';
import React from 'react';
import TestRenderer from 'react-test-renderer';
import sinon, { SinonSandbox } from 'sinon';
import spaceService from '../../services/space';
import AuthStore from '../../store/auth';
import { PARAM_SPACE } from '../../constants';
import routes from '../../routes';
import JoinPage from './JoinPage';
import { sampleSpace } from '../../test-fixtures/object-mother';

describe('JoinPage', () => {
    let sandbox: SinonSandbox;
    beforeEach((): void => {
        sandbox = sinon.createSandbox();
        jest.resetModules();
    });

    afterEach((): void => {
        sandbox.restore();
    });

    it('should join space and redirect to it', async () => {
        const fakeNavigation = {
            navigate: jest.fn(),
        };

        sandbox.stub(spaceService, 'getInvitedSpaces').resolves([sampleSpace])
        const joinSpace = sandbox.stub(spaceService, 'joinSpace').resolves();

        const testRenderer = TestRenderer.create(
            <AuthStore.Provider
                initialState={
                    {
                        isAuthenticated: true,
                        isLoading: false,
                        currentUser: {
                            id: 'abc',
                            givenName: 'Jon',
                            familyName: 'Snow',
                            email: 'jonsnow@winterfell.com',
                            avatarUrl: null,
                        },
                    } as any
                }>
                <JoinPage navigation={fakeNavigation} />
            </AuthStore.Provider>
        );
        // Await initial render
        await act(async () => {
            await new Promise(setImmediate);
        });
        // Using snapshot to assert preconditions.
        expect(testRenderer.toJSON()).toMatchSnapshot();

        // Select a space to join.
        await act(async () => {
            const selectSpace = testRenderer.root.findByProps({
                testID: `SelectSpace_${sampleSpace.id}`,
            });
            selectSpace.props.onPress();            
            await new Promise(setImmediate);
        });

        await act(async () => {
            await new Promise(setImmediate);
        });

        // Assert expected results.
        // expect(joinSpace.calledOnce).toBeTruthy();
        // expect(joinSpace.calledWith(sampleSpace)).toBeTruthy();
        // assert redirect occurrs.
        expect(fakeNavigation.navigate).toBeCalledWith(
            routes.ONBOARDING_SPACE_INTRO, {
            [PARAM_SPACE]: sampleSpace,
        });
    });
});