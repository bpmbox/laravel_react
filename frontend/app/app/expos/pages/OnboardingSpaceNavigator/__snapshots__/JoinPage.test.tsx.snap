// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JoinPage should join space and redirect to it 1`] = `
<SafeAreaView
  style={
    Object {
      "flex": 1,
    }
  }
>
  <ScrollView
    contentContainerStyle={Array []}
    keyboardDismissMode="on-drag"
    keyboardShouldPersistTaps="handled"
    scrollable={true}
    style={
      Array [
        undefined,
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 56,
          "maxWidth": "100%",
        }
      }
    >
      <View
        size={1}
        style={
          Object {
            "paddingHorizontal": 16,
            "width": "100%",
          }
        }
        title="Switch Account"
      >
        <TouchableOpacity
          activeOpacity={0.9}
          hoverStyle={
            Object {
              "borderRadius": 7.4375,
              "opacity": 1,
              "shadowColor": "#000000",
              "shadowOffset": Object {
                "height": 10,
                "width": 0,
              },
              "shadowOpacity": 0.1,
              "shadowRadius": 20,
            }
          }
          onPress={[Function]}
          onPressIn={[Function]}
          onPressOut={[Function]}
        >
          <Text
            ellipsizeMode="tail"
            numberOfLines={1}
            style={
              Object {
                "backgroundColor": "#0066FF",
                "borderColor": "#00000000",
                "borderRadius": 7.4375,
                "borderWidth": 0,
                "color": "#FFFFFF",
                "fontFamily": "GenEiGothicP-R",
                "fontSize": 17,
                "fontWeight": "500",
                "lineHeight": 42.5,
                "overflow": "hidden",
                "paddingHorizontal": 32,
                "textAlign": "center",
                "width": "100%",
              }
            }
          >
            Switch Account
          </Text>
        </TouchableOpacity>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "maxWidth": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "flex": 1,
            "flexDirection": "column",
            "flexGrow": 1,
            "justifyContent": "center",
            "maxWidth": "100%",
            "overflow": "hidden",
          }
        }
      >
        <Text
          selectable={false}
          style={
            Object {
              "color": "#000000",
              "fontFamily": "GenEiGothicP-B",
              "fontSize": 17,
              "fontWeight": "700",
              "lineHeight": 27.200000000000003,
              "maxWidth": "100%",
              "opacity": 1,
              "paddingBottom": 4,
              "paddingLeft": 16,
              "paddingRight": 16,
              "paddingTop": 16,
              "textAlign": "left",
            }
          }
        >
          Spaces you've been invited to
        </Text>
      </View>
    </View>
    <TouchableOpacity
      activeOpacity={0.95}
      disabled={false}
      onPress={[Function]}
      onPressIn={[Function]}
      onPressOut={[Function]}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "height": 100,
            "maxWidth": "100%",
          }
        }
        testID="SelectSpace_112233"
      >
        <View
          style={
            Object {
              "backgroundColor": "#00000000",
              "borderColor": "#EAEAEA",
              "borderRadius": 7,
              "borderWidth": 0.5,
              "height": 68,
              "left": 16,
              "overflow": "hidden",
              "position": "absolute",
              "width": 68,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "height": 68,
                  "width": 68,
                },
                Object {
                  "borderRadius": 7.4375,
                },
              ]
            }
          >
            <Image
              source={
                Object {
                  "uri": "http://icons.iconarchive.com/icons/limav/game-of-thrones/512/Baratheon-icon.png",
                }
              }
              style={
                Array [
                  Object {
                    "height": 68,
                    "width": 68,
                  },
                  Object {
                    "borderRadius": 7.4375,
                  },
                ]
              }
            />
          </View>
        </View>
        <View
          style={
            Object {
              "alignItems": "flex-start",
              "flex": 1,
              "flexDirection": "column",
              "flexGrow": 1,
              "justifyContent": "center",
              "maxWidth": "100%",
              "overflow": "hidden",
            }
          }
        >
          <Text
            ellipsizeMode="tail"
            numberOfLines={1}
            selectable={false}
            style={
              Object {
                "color": "#000000",
                "fontFamily": "GenEiGothicP-R",
                "fontSize": 17,
                "fontWeight": "500",
                "lineHeight": 25.5,
                "maxWidth": "100%",
                "opacity": 1,
                "paddingBottom": 4,
                "paddingLeft": 100,
                "paddingRight": 16,
                "paddingTop": 4,
                "textAlign": "left",
              }
            }
          >
            Space One
          </Text>
        </View>
      </View>
    </TouchableOpacity>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 44,
          "maxWidth": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "flex": 1,
            "flexDirection": "column",
            "flexGrow": 1,
            "justifyContent": "center",
            "maxWidth": "100%",
            "overflow": "hidden",
          }
        }
      >
        <Text
          ellipsizeMode="tail"
          numberOfLines={1}
          selectable={false}
          style={
            Object {
              "color": "#000000",
              "fontFamily": "GenEiGothicP-SL",
              "fontSize": 17,
              "fontWeight": "400",
              "lineHeight": 25.5,
              "maxWidth": "100%",
              "opacity": 1,
              "paddingBottom": 4,
              "paddingLeft": 16,
              "paddingRight": 16,
              "paddingTop": 4,
              "textAlign": "left",
            }
          }
        >
          
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "maxWidth": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "flex": 1,
            "flexDirection": "column",
            "flexGrow": 1,
            "justifyContent": "center",
            "maxWidth": "100%",
            "overflow": "hidden",
          }
        }
      >
        <Text
          selectable={false}
          style={
            Object {
              "color": "#777777",
              "fontFamily": "GenEiGothicP-SL",
              "fontSize": 14.875,
              "fontWeight": "400",
              "lineHeight": 22.3125,
              "maxWidth": "100%",
              "opacity": 1,
              "paddingBottom": 4,
              "paddingLeft": 16,
              "paddingRight": 16,
              "paddingTop": 4,
              "textAlign": "left",
            }
          }
        >
          Using account: jonsnow@winterfell.com
        </Text>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "right": 0,
          },
          Object {
            "height": 0,
          },
          undefined,
        ]
      }
    />
  </ScrollView>
</SafeAreaView>
`;
